### PR TITLE
Creating Healthcare Entities for XDM

### PR DESCRIPTION
**Pull request by mjasrotia**
Please link to the issue #…

### CODE REVIEW

File: components/fieldgroups/medication/healthcare-medication.schema.json [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "title": "Healthcare medication",
+  "type": "object",
+  "meta:extensible": true,
+  "meta:abstract": true,
+  "meta:tags": {
+    "industry": ["healthcare_services"]
+  },
+  "meta:intendedToExtend": ["https://ns.adobe.com/xdm/classes/medication"],
+  "description": "Details of a Medication.",
```
Comment (hanessian) [inline components/fieldgroups/medication/healthcare-medication.schema.json:18]: Medication details such as brand name, lot number, and quantity.

File: components/fieldgroups/plan/healthcare-plan-details.schema.json [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "title": "Healthcare Plan Details",
+  "type": "object",
+  "meta:extensible": true,
+  "meta:abstract": true,
+  "meta:tags": {
+    "industry": ["healthcare_services"]
+  },
+  "meta:intendedToExtend": ["https://ns.adobe.com/xdm/classes/plan"],
+  "description": "Details of benefits received by a patient consisting of medical care (provided directly or through insurance or reimbursement) under any hospital or medical service policy, plan contract, or HMO contract offered by a health insurance company or a group health plan.",
```
Comment (hanessian) [inline components/fieldgroups/plan/healthcare-plan-details.schema.json:18]: Patient medical plan details such as network, type, and active status.

File: components/fieldgroups/plan/healthcare-plan-details.schema.json [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+                "vision": "Vision",
+                "accident": "Accident"
+              },
+              "description": "Plan coverage type."
+            },
+            "xdm:lastVerificationDate": {
+              "title": "Last Verification Date",
+              "type": "string",
+              "format": "date-time",
+              "description": "The date on which this plan was last verified."
```
Comment (hanessian) [inline components/fieldgroups/plan/healthcare-plan-details.schema.json:109]: The date on which the plan was last verified.

File: components/fieldgroups/plan/healthcare-plan-details.schema.json [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+            },
+            "xdm:cost": {
+              "title": "Plan Cost",
```
Comment (hanessian) [inline components/fieldgroups/plan/healthcare-plan-details.schema.json:112]: remove

File: components/fieldgroups/profile/profile-healthcare-member.schema.json [hunk 5]
```diff
@@ -0,0 +1,367 @@
...
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "title": "Healthcare Member Details",
+  "type": "object",
+  "meta:extensible": true,
+  "meta:abstract": true,
+  "meta:tags": {
+    "industry": ["healthcare_services"]
+  },
+  "meta:intendedToExtend": ["https://ns.adobe.com/xdm/context/profile"],
+  "description": "Details of a person/member/subscriber who may be waiting for a care or may be receiving care or may have already received care.",
```
Comment (hanessian) [inline components/fieldgroups/profile/profile-healthcare-member.schema.json:18]: Healthcare member details of a person that has or will receive a service or care, such as contact information, primary care physician, and plan information.

File: components/fieldgroups/profile/profile-healthcare-member.schema.json [hunk 6]
```diff
@@ -0,0 +1,367 @@
...
+      "properties": {
+        "xdm:member": {
+          "title": "Member",
+          "description": "Information about the member.",
+          "type": "object",
+          "properties": {
+            "xdm:ID": {
+              "title": "Member ID",
+              "type": "string",
+              "description": "Unique Identifier of the patient/member."
```
Comment (hanessian) [inline components/fieldgroups/profile/profile-healthcare-member.schema.json:30]: Unique identifier of the patient/member.

File: components/fieldgroups/profile/profile-healthcare-member.schema.json [hunk 7]
```diff
@@ -0,0 +1,367 @@
...
+                "properties": {
+                  "xdm:providerID": {
+                    "title": "Provider ID",
+                    "type": "string",
+                    "description": "Unique identifier of the physician."
+                  },
+                  "xdm:fullName": {
+                    "title": "Full Name",
+                    "type": "string",
+                    "description": "Primary Care Physician full name."
```
Comment (hanessian) [inline components/fieldgroups/profile/profile-healthcare-member.schema.json:171]: Primary care physician's full name.

File: components/fieldgroups/profile/profile-healthcare-member.schema.json [hunk 8]
```diff
@@ -0,0 +1,367 @@
...
+                  "title": "Relationship To Member",
+                  "type": "string",
+                  "description": "Emergency contact's relationship to the member."
+                }
+              }
+            },
+            "xdm:preferredAvaiability": {
+              "title": "Preferred Availability",
+              "type": "string",
+              "description": "Patient prefered day and time availability for an appointment."
```
Comment (hanessian) [inline components/fieldgroups/profile/profile-healthcare-member.schema.json:213]: Patient's preferred day and time availability for an appointment.

File: components/fieldgroups/profile/profile-healthcare-member.schema.json [hunk 9]
```diff
@@ -0,0 +1,367 @@
...
+            },
+            "xdm:isDependent": {
+              "title": "Is Dependent",
+              "type": "boolean",
+              "description": "Indicates whether this member is a dependent."
+            },
+            "xdm:primaryMemberID": {
+              "title": "Primary Member ID",
+              "type": "string",
+              "description": "Unique Identifier of the primary subscriber if the member is dependent."
```
Comment (hanessian) [inline components/fieldgroups/profile/profile-healthcare-member.schema.json:259]: Unique identifier of the primary subscriber if the member is a dependent.

File: components/fieldgroups/profile/profile-healthcare-member.schema.json [hunk 10]
```diff
@@ -0,0 +1,367 @@
...
+                "spouse": "Spouse",
+                "child": "Child",
+                "parent": "Parent"
+              },
+              "description": "Beneficiary relationship to the subscriber if the member is a dependent (self, spouse, child, etc)."
+            },
+            "xdm:billingAccountID": {
+              "title": "Billing Account ID",
+              "type": "string",
+              "description": "Unique Identifier of the billing account."
```
Comment (hanessian) [inline components/fieldgroups/profile/profile-healthcare-member.schema.json:276]: Unique identifier of the billing account

File: components/fieldgroups/profile/profile-healthcare-member.schema.json [hunk 11]
```diff
@@ -0,0 +1,367 @@
...
+              "title": "Enrolled Plans",
+              "type": "array",
+              "items": {
+                "type": "object",
+                "additionalProperties": false,
+                "properties": {
+                  "xdm:planID": {
+                    "title": "Plan Identifier",
+                    "type": "string",
+                    "description": "Unique identifier for the enrolled plan."
```
Comment (hanessian) [inline components/fieldgroups/profile/profile-healthcare-member.schema.json:334]: Unique identifier for the plan enrolled in by the member.

File: components/fieldgroups/provider/healthcare-provider-details.schema.json [hunk 12]
```diff
@@ -0,0 +1,163 @@
...
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "title": "Healthcare Provider",
+  "type": "object",
+  "meta:extensible": true,
+  "meta:abstract": true,
+  "meta:tags": {
+    "industry": ["healthcare_services"]
+  },
+  "meta:intendedToExtend": ["https://ns.adobe.com/xdm/classes/provider"],
+  "description": "Details of an individual health professional or a health facility organization licensed to provide health care diagnosis and treatment services including medication, surgery and medical devices.",
```
Comment (hanessian) [inline components/fieldgroups/provider/healthcare-provider-details.schema.json:18]: Details of an individual health professional or a health facility organization licensed to provide health care diagnosis and treatment services.

File: components/fieldgroups/provider/healthcare-provider-details.schema.json [hunk 13]
```diff
@@ -0,0 +1,163 @@
...
+              "meta:enum": {
+                "attending_physician": "Attending Physician",
+                "physician": "Physician",
+                "emergency_physician": "Emergency Physician"
+              }
+            },
+            "xdm:practiceGroupName": {
+              "title": "Service Provider Practice Group Name",
+              "type": "string",
+              "description": "Service provider practice group Name."
```
Comment (hanessian) [inline components/fieldgroups/provider/healthcare-provider-details.schema.json:50]: Service provider practice group name

File: components/fieldgroups/provider/healthcare-provider-details.schema.json [hunk 14]
```diff
@@ -0,0 +1,163 @@
...
+            },
+            "xdm:practiceGroupType": {
+              "title": "Service Provider Practice Group Type",
+              "type": "string",
+              "description": "Service provider practice group type."
+            },
+            "xdm:classification": {
+              "title": "Service Provider Class/Category",
+              "type": "string",
+              "description": "Service provider classification based on class or category like (Patient-care) , (non-Ptaient care) etc.",
```
Comment (hanessian) [inline components/fieldgroups/provider/healthcare-provider-details.schema.json:60]: "Service provider classification based on class or category (e.g. patient-care, non-patient care, etc)."

File: components/fieldgroups/provider/healthcare-provider-details.schema.json [hunk 15]
```diff
@@ -0,0 +1,163 @@
...
+                  "xdm:address": {
+                    "title": "Provider Address",
+                    "$ref": "https://ns.adobe.com/xdm/common/address"
+                  }
+                }
+              }
+            },
+            "xdm:qualifications": {
+              "title": "Qualifications",
+              "description": "Certification, licenses, or training pertaining to the provision of care",
```
Comment (hanessian) [inline components/fieldgroups/provider/healthcare-provider-details.schema.json:121]: Certification, licenses, or training pertaining to the provision of care.

**Review by hanessian — CHANGES_REQUESTED** [review]
Added come comments mainly for description/title clean up. If you want to take the change I proposed, I'd suggest copy/pasting since there are small edits like adding periods, adding an extra letter to a word, etc, which might be hard to catch by eye.

File: components/fieldgroups/profile/profile-healthcare-member.schema.json (hunk 6 above)
Comment (mjasrotia) [reply components/fieldgroups/profile/profile-healthcare-member.schema.json:30]: We are removing this so, this change won't be required

**Pull request merged**